pytest_fixtures/component/http_proxy.py

```diff
@@ -0,0 +1,48 @@
+import pytest
+from fauxfactory import gen_string
+
+from robottelo.config import settings
+
+
+def create_http_proxy(sat, org, http_proxy_type):
+    """
+    Creates HTTP proxy.
+    :param str sat: Satellite to use.
+    :param str org: Organization
+    :param str http_proxy_type: None, False, True
+    """
+    if http_proxy_type is False:
+        return sat.api.HTTPProxy(
+            name=gen_string('alpha', 15),
+            url=settings.http_proxy.un_auth_proxy_url,
+            organization=[org.id],
+        ).create()
+    if http_proxy_type:
+        return sat.api.HTTPProxy(
+            name=gen_string('alpha', 15),
+            url=settings.http_proxy.auth_proxy_url,
+            username=settings.http_proxy.username,
+            password=settings.http_proxy.password,
+            organization=[org.id],
+        ).create()
+
+
+@pytest.fixture(scope='function')
+def function_http_proxy(request, module_manifest_org, target_sat):
```

Comment:
If more fixtures are added for diff scopes, then mention types, like `module_setup_http_proxy`
```suggestion
def setup_http_proxy(request, module_manifest_org, target_sat):
```

Reply:
updated the pr.